dialoptions.go

```diff
@@ -139,6 +139,15 @@ func newJoinDialOption(opts ...DialOption) DialOption {
 	return &joinDialOption{opts: opts}
 }
 
+// WithShareWriteBuffer allows reusing per-connection transport write buffer.
+// If this option is set to true every connection will release the buffer after
+// flushing the data on the wire.
+func WithShareWriteBuffer(val bool) DialOption {
```

Comment:
Maybe this name and the name of the `ServerOption` could do with some bikeshedding. Options that come to my mind are
- WithWriteBufferSharing
- WithSharedWriteBuffer

Reply:
done, I used WithSharedWriteBuffer